### ISSUE TITLE
Support pass-through information from the client to the target endpoint

### ISSUE THREAD

**Issue by lius** [strongloop/loopback-connector-remote#20]
In LB applications, a model backed by a datasource that uses the remote connector should have a configuration to enable use of client information (e.g.: from request) and perform a pass-through to the target endpoint, like a proxy would.

There are some use cases that would benefit, like microservices within the same boundaries that share authentication and authorisation rules. In this type of controlled environment, that there is no requirements to use OAuth2 (due to all applications deals with resources that belongs to the same owners list, and resource server belongs to the same maintainer as authorization and client application servers), this functionally would be a more "direct-way" to build seamless interconnected distributed applications, specially for scenarios where is not necessary to implement a API Gateway.

In terms of LB concepts, I think this feature could be a mechanism to spread (serialise and deserialise, since both sides are LB applications) information provided by context middleware, making the entire stack work as expected on the remote server side, include business rules the access context information but are isolated from web code (req, res, express middlewares, etc).

**Comment by bajtos** [comment]
+1

@ritch @kraman is there any workaround until we implement a full solution? E.g. using before-remote hooks.

**Comment by ritch** [comment]
A snippet demonstrating the desired behavior would be helpful to clarify exactly what you're asking for.

**Comment by lius** [comment]
@ritch Ok. I made the diagram below to help describe the behavior. Probably passing more information than access token would be more useful. I'm not sure about this, but there is a context concept implemented in context middleware that can be used for that purpose.

@bajtos Yes, I'm using before-remote hooks all around models backed by a remote connector to push access token in some way that it goes forward on the request to the second application.

![20150609-remote-token-passing](https://cloud.githubusercontent.com/assets/62179/8056599/3be72eac-0e80-11e5-8768-90959a07f0d4.png)

**Comment by lius** [comment]
Guys, I'm trying to figure out how to overcome the problem, but I need some guidance, because I'm not sufficient familiar with strong-remoting code. So my first step was investigate the code.

First, I understood that strong-remoting maps a receiving request to a model shared method, and then, the connector is responsible to finish the job. During the first step (receiving and mapping), strong-remoting is aware of ctx/context middleware (access token, etc), but after call the method on SharedMethod.prototype.invoke, all underling calls don't have access to that info.

Since remote connector is implemented using strong-remoting, I thought the awareness to context should be available, but I did not manage to make it work this way.

I think the steps are understood (strong-remoting -> connector-remote -> rest-adapter -> strong-remoting -> http createRequest-> execStack -> http invoke), but I need some suggestions on what to try next.

Thanks!

**Comment by lius** [comment]
Any news related to this feature? Thanks!

**Comment by DylanSale** [comment]
+1 I also need this functionality

**Comment by el-davo** [comment]
+1 this would be very handy

**Comment by aniljaangra** [comment]
Anything on this?
How Do i pass through a acessToken on the first place  for every request...

**Comment by lius** [comment]
@aniljaangra Manually, from client to server or server to server, just pass the token using Authorization header on your HTTP requests. The purpose of this issue is for automatically on server to server scenario.

**Comment by aniljaangra** [comment]
yes i am talking about server to server only... users hit server with access-token in header ... server then hits another server via remote-connector... i want to pass the access-token to the remote-server...
